/*===------------------------------------------------------------------------------*-
proto -*-===*\
 *                          _
 *                         | |
 *                       __| | __ ___      ___ ___
 *                      / _` |/ _` \ \ /\ / / '_  |
 *                     | (_| | (_| |\ V  V /| | | |
 *                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
 *
 *
 *  This file is distributed under the MIT License (MIT).
 *  See LICENSE.txt for details.
 *
\*===------------------------------------------------------------------------------------------===*/

// @defgroup sir_proto SIR protobuf
// @brief This is a SIR description for Google's protocol buffers library
// <a href="https://developers.google.com/protocol-buffers/">Protobuf</a>.

syntax = "proto3";
package dawn.proto.enums;

// Don't use the 0 value, because it's the default and doesn't get serialized,
// see https://developers.google.com/protocol-buffers/docs/proto3#default
enum GridType {
  GridTypeUnknown = 0; // don't use!
  Unstructured = 1;
  Cartesian = 2;
}

enum LocationType {
  LocationTypeUnknown = 0;
  Vertex = 1;
  Cell = 2;
  Edge = 3;
}
